### ISSUE TITLE
ImportError: libSM.so.6: cannot open shared object file: No such file or directory

### ISSUE THREAD

**Issue by jcohenadad** [neuropoly/spinalcordtoolbox#1784]
On joplin (Ubuntu) after fresh git clone of master/fc72f34d4251de09d3e92071c66b40ac7fe4c92a and while running `batch_processing`:
~~~
Label discs...

Generate output files...
  File created: ./t2_seg_labeled.nii.gz
  File created: ./t2_seg_labeled_discs.nii.gz
  File created: ./warp_curve2straight.nii.gz
  File created: ./warp_straight2curve.nii.gz
  File created: ./straight_ref.nii.gz

Remove temporary files...
rm -rf /tmp/sct-180527181207-label_vertebrae-95hPlw
Traceback (most recent call last):
  File "/home/jcohen/code/sct-release/scripts/sct_label_vertebrae.py", line 953, in <module>
    main()
  File "/home/jcohen/code/sct-release/scripts/sct_label_vertebrae.py", line 361, in main
    generate_qc(fname_in, labeled_seg_file, args, path_qc)
  File "/home/jcohen/code/sct-release/scripts/sct_label_vertebrae.py", line 370, in generate_qc
    import spinalcordtoolbox.reports.qc as qc
  File "/home/jcohen/code/sct-release/python/lib/python2.7/site-packages/spinalcordtoolbox/reports/qc.py", line 19, in <module>
    import matplotlib.pyplot as plt
  File "/home/jcohen/code/sct-release/python/lib/python2.7/site-packages/matplotlib/pyplot.py", line 114, in <module>
    _backend_mod, new_figure_manager, draw_if_interactive, _show = pylab_setup()
  File "/home/jcohen/code/sct-release/python/lib/python2.7/site-packages/matplotlib/backends/__init__.py", line 32, in pylab_setup
    globals(),locals(),[backend_name],0)
  File "/home/jcohen/code/sct-release/python/lib/python2.7/site-packages/matplotlib/backends/backend_qt4agg.py", line 18, in <module>
    from .backend_qt5agg import FigureCanvasQTAggBase as _FigureCanvasQTAggBase
  File "/home/jcohen/code/sct-release/python/lib/python2.7/site-packages/matplotlib/backends/backend_qt5agg.py", line 15, in <module>
    from .backend_qt5 import QtCore
  File "/home/jcohen/code/sct-release/python/lib/python2.7/site-packages/matplotlib/backends/backend_qt5.py", line 31, in <module>
    from .qt_compat import QtCore, QtGui, QtWidgets, _getSaveFileName, __version__
  File "/home/jcohen/code/sct-release/python/lib/python2.7/site-packages/matplotlib/backends/qt_compat.py", line 124, in <module>
    from PyQt4 import QtCore, QtGui
ImportError: libSM.so.6: cannot open shared object file: No such file or directory
~~~

Maybe due to the impossibility of opening a figure (connected via ssh without -X)

**Comment by jcohenadad** [comment]
I confirm this is due to the QC module trying to open a matplotlib figure. I tried v3.1.1 on the same station and there is no error during QC generation. I am wondering if something related to Agg has been since the last release? @zougloub do you have any insight?

additional notes: a notable difference with v3.1.1 is that previously, the full html was not created, only the json was produced, and then sct_qc was run, pointing to the qc folder. However, even sct_qc (from v3.1.1) works fine in the tested station (ssh without -x).

**Comment by zougloub** [comment]
The import error is due to:
- libsm6 (a basic X11 dependency) not being present on the machine (when Conda's PyQt assumes it is) which is headless
- the default chosen backend by pyplot being QtAgg

Short-term workaround: when working on a headless machine, `export MPLBACKEND=Agg` and this problem shouldn't happen.
Proper solution: do not use pyplot, rather use matplotlib OOP API, and select the backend by hand.

**Comment by jcohenadad** [comment]
Indeed, `export MPLBACKEND=Agg` did the trick. Thanks!

**Comment by joshuacwnewton** [comment]
This is very similar to #2005: it's an issue related to `MPLBACKEND` that was addressed in #2238, but the issue was never closed. So, closing now. :)

**Comment by kousu** [comment]
I did a quick test (with docker, not a full system) and I believe this has been fixed upstream, so we don't need to think about `MPLBACKEND` anymore:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3401#issuecomment-861614835

(and we shouldn't because we should be respecting [the three ways users can choose to use matplotlib](https://matplotlib.org/stable/tutorials/introductory/usage.html#selecting-a-backend)

> 1.    The rcParams["backend"] parameter in your matplotlibrc file
> 2.    The MPLBACKEND environment variable
> 3.   The function matplotlib.use()

)
